File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        maxi = 0
        for x in s:
            if x-1 not in s:
                l = 1
                while x + 1 in s:
                    l += 1
                    x += 1
                    
                maxi = max(maxi, l)
                
        return maxi
    
    

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        unordered_set<int> s;
        for (auto n: nums){
            s.insert(n);
        }
        int maxi = 1;
        for (auto n: nums) {
            if (s.find(n-1) == s.end()){
                int cnt = 1;
                int g = n+1;
                while (s.find(g) != s.end()) g++, cnt++;
                maxi = max(maxi, cnt);
            }
        }
        return maxi;
    }
};